edx/cookiecutter-django-app
# edx-enterprise improvements

Another bunch of updates from edx-enterprise:

* Introduced project_title variable - used as the human-readable project title (in docs) and as a source to generate repo/app name.
* Moved PR template to where github expects it to be
* Updated Changelog format so it generates correct changelog ToC
* Added section markers to i18n and testing - they proved to be useful when actually writing documentation
* Added improvements and fixes to conf.py resulted from trying to submit the docs to readthedocs
* suppressed pylinting docs/conf.py as it is problematic to fix it (pylint runs in tox env, where importing the app locally does not work)

Testing instructions:

1. Run tests (aka `make validate`)
2. Try baking the project with different settings and make sure everything works

**Author concerns:**

I had to suppress pylinting the `docs/conf.py` because it now relies on importing the app locally, while pylint runs from different working dir. If there are a better way to solve it than excluding conf.py - I'd really love to turn it back on.

The reason behind importing the project locally is twofold:

1. conf.py sets versions to current app version (otherwise they are never updated in practice)
2. It might be necessary to actually build the docs for readthedocs (@mattdrayer - I'd appreciate your comments on this).
